pkg/sync/syncer.go

```diff
@@ -745,22 +814,91 @@ func (s *syncer) syncGrantsForResource(ctx context.Context, resourceID *v2.Resou
 	if err != nil {
 		return err
 	}
-	for _, grant := range resp.List {
+
+	nextPageToken := resp.NextPageToken
+
+	respAnnos := annotations.Annotations(resp.GetAnnotations())
+	etagMatch := respAnnos.Contains(&v2.ETagMatch{})
+
+	var grantsRet []*v2.Grant
+
+	// We have a previous etag, and the connector has indicated an etag match
+	if etagMatch {
+		if prevEtag == nil {
+			return errors.New("connector returned an etag match - but there is no previous sync generation to use")
+		}
```

Comment:
Should this err if next page token is also set? should it set next page token to empty string? What does it mean if there's an etag match and a next page token? Code below starts with empty page string and iterates through all previous request pages. I see the okta connector code returns an empty page string when etagmatch is being returned.

Reply:
The code below is making a request against the store, and not the connector, so it is paginating through all of the grants from the previous generation.

I could go either way on if this is an error or not. As implemented it should still work if additional pages are provided, but we won't do the work to see if there is a previous generation for the resource on additional pages.